noir-projects/aztec-nr/aztec/src/utils/bytes/bounded_vec.nr

```diff
@@ -0,0 +1,324 @@
+/// This file is largely a copy of the `array.nr` file, but with the `BoundedVec` type instead of the `[u8; N]` type.
+/// Some functions (e.g. get_random_bytes) from `array.nr` are missing though as they were not yet needed and I am not
+/// sure if they ever will be.
+
+/// Converts the input bytes into a BoundedVec of fields. A Field is ~254 bits meaning that each field can store 31
+/// whole bytes.
+///
+/// The input bytes are chunked into chunks of 31 bytes. Each 31-byte chunk is viewed as big-endian, and is converted
+/// into a Field. If the last chunk is less than 31 bytes long, then only the relevant bytes are considered.
+/// For example, [1, 10, 3] is encoded as [1 * 256^2 + 10 * 256 + 3]
+/// Note: ceil(N / 31) = (N + 30) / 31
+pub fn be_bytes_31_to_fields<let N: u32>(
+    bytes: BoundedVec<u8, N>,
+) -> BoundedVec<Field, (N + 30) / 31> {
```

Comment:
This function is equivalent to the array version except that in the code we iterate over bytes.len() instead of N.